homeassistant/components/mysensors/device.py

```diff
@@ -19,33 +32,99 @@
 MYSENSORS_PLATFORM_DEVICES = "mysensors_devices_{}"
 
 
-def get_mysensors_devices(hass, domain):
-    """Return MySensors devices for a platform."""
-    if MYSENSORS_PLATFORM_DEVICES.format(domain) not in hass.data:
-        hass.data[MYSENSORS_PLATFORM_DEVICES.format(domain)] = {}
-    return hass.data[MYSENSORS_PLATFORM_DEVICES.format(domain)]
-
-
 class MySensorsDevice:
     """Representation of a MySensors device."""
 
-    def __init__(self, gateway, node_id, child_id, name, value_type):
+    def __init__(
+        self,
+        gateway_id: GatewayId,
```

Comment:
Isn't the gateway id just a string?

Reply:
It used to be `int` when the id was literally `id(gateway)`. I changed it to `str`.

The special type allows checkers to find some simple bugs and serves as a reminder of what you are dealing with.
The type is especially helpful when nested in tuples or dicts.

The type `DevId` was also added for the same reason.